provider/keycloak_openid_user_attribute_protocol_mapper_test.go

```diff
@@ -0,0 +1,329 @@
+				Check:  testKeycloakOpenIdUserAttributeProtocolMapperExists(resourceName),
+			},
+		},
+	})
+}
+
+func testAccKeycloakOpenIdUserAttributeProtocolMapperDestroy() resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+
+		for resourceName, rs := range state.RootModule().Resources {
+			if rs.Type != "keycloak_openid_user_attribute_protocol_mapper" {
+				continue
+			}
+
+			mapper, _ := getUserAttributeMapperUsingState(state, resourceName)
+
+			if mapper != nil {
+				return fmt.Errorf("openid user attribute protocol mapper with id %s still exists", rs.Primary.ID)
+			}
+		}
+
+		return nil
+	}
+}
+
+func testKeycloakOpenIdUserAttributeProtocolMapperExists(resourceName string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+		_, err := getUserAttributeMapperUsingState(state, resourceName)
+
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func getUserAttributeMapperUsingState(state *terraform.State, resourceName string) (*keycloak.OpenIdUserAttributeProtocolMapper, error) {
+	rs, ok := state.RootModule().Resources[resourceName]
+
+	if !ok {
+		return nil, fmt.Errorf("resource not found in TF state: %s ", resourceName)
+	}
+	id := rs.Primary.ID
+	realm := rs.Primary.Attributes["realm_id"]
+	clientId := rs.Primary.Attributes["client_id"]
+	clientScopeId := rs.Primary.Attributes["client_scope_id"]
+
+	keycloakClient := testAccProvider.Meta().(*keycloak.KeycloakClient)
+
+	if clientId != "" {
+		return keycloakClient.GetOpenIdUserAttributeProtocolMapperForClient(realm, clientId, id)
+	} else {
+		return keycloakClient.GetOpenIdUserAttributeProtocolMapperForClientScope(realm, clientScopeId, id)
+	}
+}
+
+func testKeycloakOpenIdUserAttributeProtocolMapper_basic_client(realmName, clientId, mapperName string) string {
+	return fmt.Sprintf(`
+resource "keycloak_realm" "realm" {
+	realm = "%s"
+}
+
+resource "keycloak_openid_client" "openid-client" {
+	realm_id = "${keycloak_realm.realm.id}"
+	client_id = "%s"
+}
+
+resource "keycloak_openid_user_attribute_protocol_mapper" "user-attribute-mapper-client" {
+  	name = "%s"
+	realm_id = "${keycloak_realm.realm.id}"
+  	client_id = "${keycloak_openid_client.openid-client.id}"
+  	user_attribute = "foo"
+  	claim_name = "bar"
+}`, realmName, clientId, mapperName)
+}
+
+func testKeycloakOpenIdUserAttributeProtocolMapper_basic_clientScope(realmName, clientScopeId, mapperName string) string {
+	return fmt.Sprintf(`
+resource "keycloak_realm" "realm" {
+	realm = "%s"
+}
+
+resource "keycloak_openid_client_scope" "client-scope" {
+  name                = "%s"
+  realm_id            = "${keycloak_realm.realm.id}"
+}
+
+resource "keycloak_openid_user_attribute_protocol_mapper" "user-attribute-mapper-client-scope" {
+  	name = "%s"
+	realm_id = "${keycloak_realm.realm.id}"
+  	client_scope_id = "${keycloak_openid_client_scope.client-scope.id}"
+  	user_attribute = "foo"
+  	claim_name = "bar"
+}`, realmName, clientScopeId, mapperName)
```

Comment:
Can you use tabs for indentation and align properties on `=` for all tf config?  I just opened #30 because other acceptance tests I wrote also don't do this, and I'd like it to be consistent.